src/collector/Group.cpp

```diff
@@ -334,26 +338,14 @@ void Group::update_status(bool forbidden_dht)
             if (backend_ts < cur_ts)
                 backend_ts = cur_ts;
 
-            if (b_status == Backend::BAD) {
-                have_bad = true;
-                break;
-            } else if (b_status == Backend::RO) {
+            if (b_status == Backend::RO) {
                 have_ro = true;
             } else if (b_status != Backend::OK) {
                 have_other = true;
             }
         }
```

Comment:
After 'for' both `have_ro` and `have_other` can be `true`, but code below, looks like, doesn't expect that both of them can be `true`.